sunbird_service_api_test/src/test/java/org/sunbird/integration/test/location/LocationTest.java

```diff
@@ -0,0 +1,368 @@
+package org.sunbird.integration.test.location;
+
+import com.consol.citrus.annotations.CitrusTest;
+import com.consol.citrus.context.TestContext;
+import com.consol.citrus.dsl.builder.HttpClientActionBuilder.HttpClientReceiveActionBuilder;
+import com.consol.citrus.http.client.HttpClient;
+import com.consol.citrus.testng.CitrusParameters;
+import com.consol.citrus.validation.json.JsonMappingValidationCallback;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.HttpStatus;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.response.ResponseCode;
+import org.sunbird.common.util.Constants;
+import org.sunbird.integration.test.common.BaseCitrusTest;
+import org.sunbird.integration.test.user.EndpointConfig.TestGlobalProperty;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * Class contains functional test cases for location APIs.
+ * @author arvind.
+ */
+public class LocationTest extends BaseCitrusTest {
+
+
+  private static String stateLocationId = null;
+
+  private static String districtLocationId = null;
+  private static final String CREATE_LOCATION_URI = "/v1/location/create";
+  private static final String UPDATE_LOCATION_URI = "/v1/location/update";
+  private static final String DELETE_LOCATION_URI = "/v1/location/delete";
+  private static final String LOCATION_TEMPLATE_PATH = "templates/location/create/";
+  private static final String LOCATION_TEMPLATE_PATH_UPDATE = "templates/location/update/";
+  private static final String LOCATION_TEMPLATE_PATH_DELETE = "templates/location/delete/";
+
+  private static final String STATE_CODE= "State-02";
+  private static final String STATE_NAME= "State-0001-name";
+  private static final String DISTRICT_CODE="District-02";
+  private static final String DISTRICT_NAME="District-0001-name";
+  private static Stack<String> stack = new Stack();
+
+
+  @Autowired
+  private HttpClient restTestClient;
+  @Autowired
+  private TestGlobalProperty initGlobalValues;
+  private ObjectMapper objectMapper = new ObjectMapper();
+
+  @DataProvider(name = "createStateLocationDynamicDataProvider")
+  public Object [] [] createStateLocationDynamicDataProvider () {
+    return new Object [] [] {
+        new Object[] { createStateLocationMap(), LOCATION_TEMPLATE_PATH +"createLocationSuccessResponse.json", "createState" },
+        new Object[] { createStateLocationMap(), LOCATION_TEMPLATE_PATH +"createLocationFailureResponseForDuplicateCode.json", "createLocationStateWithDupicateCode" },
```

Comment:
change createLocationStateWithDupicateCode  to "createStateWithDuplicateCode" 
change  Dupicate to Duplicate